### DISCUSSION TITLE
Import markdown folder to a single file wiki

### DISCUSSION THREAD

**Discussion by saccarosium** [TiddlyWiki/TiddlyWiki5#7307] [Q&A] [answered]
Hi,
I'm tring to switch to tiddlywiki from my folder full of markdown, and I want to adopt the mediawikish syntax of tiddlywiki.
I can't seem to find a solution to this.

**Comment by pmario** [comment, marked as answer]
How does 1 of your markdown files look like? ... 

TW has 2 types of tiddlers that are easy to be imported. 

1. my-content.tid file
2. my-content.tid + my-content.tid.meta file

add 1) the file needs to look like this: 

```
title: title of the tiddler to be displayed
tags: tag1 [[tag with spaces]]
any-field: extra info that may be needed
type: text/markdown

your text content comes here. 
```

If a tiddler is created with the wiki and has a type-field `text/markdown` or `text/x-markdown` as shown above it will be created as 2 fieles, if used with the node.js server. 

Usually working with `.tid` files is easer in TW, because you only have to deal with 1 file. ... But ... 

add 2) Additional meta info

If you already have a lot of `.md` files, you only need to create `.md.meta` files to be able to import them into TW, without the need to change the `.md` file. 

`my-content.md.meta` would look as follows. .. So it basically gets the stand alone header info of the .tid file. 

```
title: title of the tiddler to be displayed
tags: tag1 [[tag with spaces]]
any-field: extra info that may be needed
type: text/markdown
```

Hope that helps

**Reply by saccarosium** [reply]
> How does 1 of your markdown files look like? ...

Structure wise I usually go for the following structure:
```
# Name of the file

## Sections

#tags
```

For the flavor of markdown I'm working on is `commonmark` plus some extentions from github flavored markdown.

I'm not really understanding how exacly you import a file... Do you drang and drop in the wiki? Isn't the wiki only a html file? If so why I need `.meta` file?

> ```
> title: title of the tiddler to be displayed
> tags: tag1 [[tag with spaces]]
> any-field: extra info that may be needed
> type: text/markdown
>
> your text content comes here. 
> ```

Ok so there is a way to convert markdown, or html, files in this format?

**Comment by pmario** [comment]
Sorry for the confusion. ... 

>I'm not really understanding how exacly you import a file... Do you drang and drop in the wiki?

Yes that's possible. If you drag a test.md file from a folder to the wiki it will be imported. I just wanted to know if you use the "matter most"  md extension, which also has some meta data at the top of the .md file. .. 

Since that's not the case you don't need a meta file. 

If you drag & drop import some .md files from you folder to eg: https://tiddlywiki.com/plugins/tiddlywiki/markdown/ you can play around with your files. 

> Isn't the wiki only a html file? If so why I need .meta file?

The meta files come in handy, if you use a TW Client-Server configuration. Since you seem to use a single file wiki you don't need one.

**Reply by pmario** [reply]
With the upcoming TW v5.2.6 we did move to a different markdown parser library. So you may also have a look at: https://tiddlywiki.com/prerelease/plugins/tiddlywiki/markdown/ (see the prerelease) in the URL. .. 

It has a slightly different behaviour and should be overall more powerful.

**Reply by saccarosium** [reply]
If I wanna use the tiddly wiki syntax, I have to convert every markdown file by hand, or there is a converter (like pandoc).

**Reply by pmario** [reply]
As far as I know, there is no automatic converter. The differences are not too big anymore. .. It depends a bit how your wikitext looks like. 

You could do the conversion step by step. Whenever you edit a MD tiddler you convert it. In the mean time you can parse them using the markdown-plugin. 

If you want to convert them immediately. There are some examples 

**MD -> TW**

```
Headings 

#  -> !
## -> !!
...

Unordered Lists

- level 1    -> * level 1
  - level 2  -> ** level 2

Ordered Lists

1. level 1  -> # level 1
   1.1. level 2 -> ## level 2

>Quotes are the same

<<<
TW has it's own blockquote 
<<<

    ```
    code blocks are the same
    ```
```

Formatting like bold is different: https://tiddlywiki.com/#Formatting%20in%20WikiText

Wikitext Overview: https://tiddlywiki.com/#WikiText

**Reply by saccarosium** [reply]
Thank  for your help, you have been very helpfull.

**Comment by linonetwo** [comment]
There is a https://github.com/tiddly-gittly/md-to-tid , also on the CPL https://tw-cpl.netlify.app/
